libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -771,6 +771,53 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	{
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		solAssert(!functionType->gasSet(), "Gas limit set for contract creation.");
+		solAssert(
+			functionType->returnParameterTypes().size() == 1,
+			"Constructor should return only one type"
+		);
+
+		TypePointers argumentTypes;
+		string constructorParams;
+		for (ASTPointer<Expression const> const& arg: arguments)
+		{
+			argumentTypes.push_back(arg->annotation().type);
+			constructorParams += ", " + IRVariable{*arg}.commaSeparatedList();
+		}
+
+		ContractDefinition const* contract =
+			&dynamic_cast<ContractType const&>(*functionType->returnParameterTypes().front()).contractDefinition();
+		m_context.objects().insert(contract);
```

Comment:
`objects()` is the contracts that this contract references or creates, right?

Reply:
Should be the list of all contracts that are created inside of this contract with new

Reply:
Can this somehow be reflected in the name? :)

Reply:
I was thinking subObjects was enough, but maybe subObjectsCreated?